include/exec/sequence/transform_each.hpp

```diff
@@ -0,0 +1,161 @@
+        }
+
+        template <same_as<get_env_t> _GetEnv, __decays_to<__t> _Self>
+        friend env_of_t<_Receiver> tag_invoke(_GetEnv, _Self&& __self) noexcept {
+          return stdexec::get_env(__self.__op_->__receiver_);
+        }
+      };
+    };
+
+    template <class _Sender, class _ReceiverId, class _Adaptor>
+    struct __operation {
+      using _Receiver = stdexec::__t<_ReceiverId>;
+
+      struct __t : __operation_base<_Receiver, _Adaptor> {
+        subscribe_result_t<_Sender, stdexec::__t<__receiver<_ReceiverId, _Adaptor>>> __op_;
+
+        __t(_Sender&& __sndr, _Receiver __rcvr, _Adaptor __adaptor)
+          : __operation_base<
+            _Receiver,
+            _Adaptor>{static_cast<_Receiver&&>(__rcvr), static_cast<_Adaptor&&>(__adaptor)}
+          , __op_{exec::subscribe(
+              static_cast<_Sender&&>(__sndr),
+              stdexec::__t<__receiver<_Receiver, _Adaptor>>{this})} {
+        }
+
+        friend void tag_invoke(start_t, __t& __self) noexcept {
+          stdexec::start(__self.__op_);
+        }
+      };
+    };
+
+    template <class _SenderId, class _Adaptor>
+    struct __sender {
+      using _Sender = stdexec::__t<_SenderId>;
+
+      struct __t {
+        using is_sender = sequence_tag;
+        using __id = __sender;
+
+        _Sender __sender_;
+        _Adaptor __adaptor_;
+
+        template <class _Self, class _Env>
+        using __some_sender = __copy_cvref_t<
+          _Self,
+          __sequence_sndr::__unspecified_sender_of<
+            completion_signatures_of_t<__copy_cvref_t<_Self, _Sender>, _Env>>>;
+
+        template <class _Self, class _Env>
+        using completion_sigs_t = completion_signatures_of_t<
+          __call_result_t<_Adaptor&, __some_sender<_Self, _Sender>>,
+          _Env>;
+
+        template <__decays_to<__t> _Self, receiver _Receiver>
+        requires sequence_receiver_of<_Receiver, completion_sigs_t<_Self, env_of_t<_Receiver>>>
+              && sequence_sender_to<
+                   __copy_cvref_t<_Self, _Sender>,
+                   stdexec::__t<__receiver<stdexec::__id<_Receiver>, _Adaptor>>>
+        friend auto tag_invoke(subscribe_t, _Self&& __self, _Receiver __rcvr) -> stdexec::__t<
+          __operation<__copy_cvref_t<_Self, _Sender>, stdexec::__id<_Receiver>, _Adaptor>> {
+          return {
+            static_cast<_Self&&>(__self).__sender_,
+            static_cast<_Receiver&&>(__rcvr),
+            static_cast<_Self&&>(__self).__adaptor_};
+        }
+
+        template <__decays_to<__t> _Self, class _Env>
+        friend auto tag_invoke(get_completion_signatures_t, _Self&&, _Env&&)
+          -> completion_sigs_t<_Self, _Env>;
+
+        friend env_of_t<_Sender> tag_invoke(get_env_t, const __t& __self) noexcept {
+          return stdexec::get_env(__self.__sender_);
+        }
+      };
+    };
+
+    struct transform_each_t {
+      template <sender _Sender, class _Adaptor>
```

Comment:
I think "`_Adaptor`" here is a sender adaptor closure, right? You can use `stdexec::__sender_adaptor_closure` to constrain the adaptor argument (or `stdexec::__sender_adaptor_closure_for` if you know the type of the sender that will be passed to the adaptor).

Reply:
Yes I have missed this concrete vocabulary. Thank you